app/views/handlers/submission.py

```diff
@@ -47,12 +55,20 @@ def submit_questionnaire(self):
         self._questionnaire_store.save()
 
     def get_payload(self):
-        payload = convert_answers(
-            self._schema,
-            self._questionnaire_store,
-            self._full_routing_path,
-            self.submitted_at,
-        )
+        if self._metadata["version"] is AuthPayloadVersion.V2.value:
+            payload = convert_answers_v2(
+                self._schema,
+                self._questionnaire_store,
+                self._full_routing_path,
+                self.submitted_at,
+            )
+        else:
+            payload = convert_answers(
```

Comment:
Same as feedback comment, instead of have 2 calls that are duplicated. Get the function you need and call it once

Reply:
Updated 👍